Keywords/webUtils/WebTableDataComparison.groovy

```diff
@@ -0,0 +1,209 @@
+						Map<String, MapDifference.ValueDifference<String>> entriesDiffering = diff.entriesDiffering();
+						Map<String, String> entriesOnlyOnLeft = diff.entriesOnlyOnLeft();
+						Map<String, String> entriesOnlyOnRight = diff.entriesOnlyOnRight();
+						reportMapDifference(primaryKeyName, actRecordKey, entriesDiffering, entriesOnlyOnLeft,
+								entriesOnlyOnRight);
+					} else {
+						KeywordUtil.markPassed("All values matched for " + primaryKeyName + " :: " + actRecordKey)
+					}
+				}
+			}
+		} catch (Exception e) {
+			e.printStackTrace();
+			KeywordUtil.markFailed("Exception in compareActExpMap, unable to compare WebTable Map records")
+		}
+	}
+
+	/**
+	 * Get the Expected WebTable Map records missing from Actual WebTable Map
+	 * @param PrimaryKey of WebTable
+	 * @param Actual WebTable Map object
+	 * @param Expected WebTable Map object
+	 */
+	public static void getExpRecordMissingFrmActData(String primaryKeyName, Map<String, Map<String, String>> actMap,
+			Map<String, Map<String, String>> expMap) {
+		// Report records present in expected data but missing in actual data
+		for (String expRecordKey : expMap.keySet()) {
+			if (!actMap.containsKey(expRecordKey)) {
+				KeywordUtil.markFailed("Searched expected record not found in actual data :: " + primaryKeyName + " :: "
+						+ expRecordKey);
+			}
+		}
+	}
+
+	/**
+	 * Report the WebTable Map records differences
+	 * @param Key Name of WebTable record
+	 * @param Key Value of WebTable record
+	 * @param Recordentries differing Map object
+	 * @param Recordentries differing Right WebTable Map object
+	 * @param Recordentries differing of Left WebTable Map object
+	 */
+	public static void reportMapDifference(String recordKeyName, String recordKeyValue,
+			Map<String, MapDifference.ValueDifference<String>> entriesDiffering, Map<String, String> entriesOnlyOnLeft,
+			Map<String, String> entriesOnlyOnRight) {
+		try {
+			String failureMsg = "Value mismatch for field :: ";
+			for (String keyName : entriesOnlyOnLeft.keySet()) {
+				String actValue = entriesOnlyOnLeft.get(keyName);
+				String expValue = " ** NOT FOUND ** ";
+				KeywordUtil.markFailed("Failed " + recordKeyName + " :: " + recordKeyValue + " " + actValue + " " + expValue)
+			}
+			for (String keyName : entriesOnlyOnRight.keySet()) {
+				String actValue = " ** NOT FOUND ** ";
+				String expValue = entriesOnlyOnRight.get(keyName);
+				KeywordUtil.markFailed("Failed " + recordKeyName + " :: " + recordKeyValue + " " + actValue + " " + expValue)
+			}
+			for (String keyName : entriesDiffering.keySet()) {
+				String actValue = entriesDiffering.get(keyName).leftValue();
+				String expValue = entriesDiffering.get(keyName).rightValue();
+				KeywordUtil.markFailed("Failed " + recordKeyName + " :: " + recordKeyValue + " " + actValue + " " + expValue)
+			}
+		} catch (Exception e) {
+			e.printStackTrace();
+			KeywordUtil.markFailed("Exception in reportMapDifference, unable to find differences in WebTables")
```

Comment:
WebUI.comment("Exception :: " + e.getMessage());
KeywordUtil.markFailedAndStop("Exception in reportMapDifference");

Reply:
Updated exception handling as suggested